core/src/test/java/com/novoda/merlin/receiver/ConnectivityReceiverShould.java

```diff
@@ -48,19 +55,20 @@ public void notNotifyTheMerlinServiceOnNullIntents() throws Exception {
 
     @Test
     public void notNotifyTheMerlinServiceOnNonConnectivityIntents() throws Exception {
-        Intent intent = new Intent();
-        intent.setAction("notNotifyTheMerlinServiceOnNonConnectivityIntents");
```

Comment:
this test passed already but it was coincidentally passing:

```java
@Override
public void onReceive(Context context, Intent intent) {
    if (intent != null && connectivityAction(intent)) {
        boolean isConnected = getIsConnected(context, intent);
        String info = intent.getStringExtra(ConnectivityManager.EXTRA_EXTRA_INFO);
        String reason = intent.getStringExtra(ConnectivityManager.EXTRA_REASON);
        notifyMerlinService(context, new ConnectivityChangeEvent(isConnected, info, reason));
    }
}
```

It never entered the block because intent was null. Dunno how that works (i.e. we used to do `new Intent()`) would the testable android jar return null for that? :confused: